clients/client-topia/src/controllers/Visitor.ts

```diff
@@ -17,17 +19,18 @@ import { ResponseType } from "types";
  * await new Visitor(topia, id, urlSlug, { attributes: { moveTo: { x: 0, y: 0 } } });
  * ```
  */
-export class Visitor extends SDKController implements VisitorInterface {
+export class Visitor extends User implements VisitorInterface {
   readonly id: number;
   urlSlug: string;
+  user?: User;
 
   constructor(
     topia: Topia,
     id: number,
     urlSlug: string,
     options: VisitorOptionalInterface = { attributes: {}, credentials: {} },
   ) {
-    super(topia, options.credentials);
+    super(topia, { credentials: options.credentials });
```

Comment:
we should pass visitorId and urlSlug into the `User` class via `super` here so that the `User` is initialized with an initialized identity

my question is: if and when people take the initialized identity and hit a validation endpoint but get an invalid response (user is invalid) what should we do with the class instance? Nothing? Public API responsible for continuously rejecting bad creds i guess.

Reply:
What would the user class do with visitorId? The visitor class is responsible for fetching visitor info.

Reply:
A Visitor is a User. The User can be identified with playerId/urlSlug now from the `get-visitors` endpoint.

You still should be able to fetch a `User` based on who the `Visitor` is.